classes/class-wc-connect-compatibility-wc30.php

```diff
@@ -103,70 +115,110 @@ public function get_parent_product_id( WC_Product $product ) {
 		 * This is useful when an order has a product which was later deleted from the
 		 * store.
 		 *
-		 * @param int      $product_id Product ID or variation ID
-		 * @param WC_Order $order
+		 * @param int      $product_id Product ID or variation ID.
+		 * @param WC_Order $order WC Order.
+		 *
 		 * @return string The product (or variation) name, ready to print
 		 */
 		public function get_product_name_from_order( $product_id, $order ) {
-			foreach ( $order->get_items() as $line_item ) {
-				$line_product_id   = $line_item->get_product_id();
-				$line_variation_id = $line_item->get_variation_id();
+			$line_item = $this->get_line_item_from_order( $product_id, $order );
 
-				if ( ! $line_product_id ) {
-					$line_product_id = (int) get_metadata( 'order_item', $line_item->get_id(), '_product_id', true );
-				}
-
-				if ( ! $line_variation_id ) {
-					$line_variation_id = (int) get_metadata( 'order_item', $line_item->get_id(), '_variation_id', true );
```

Comment:
Part of the HPOS recipe is to remove direct DB access such as `get_metadata` (https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#auditing-the-code-base-for-direct-db-access-usage). 

Looks like this is what we are doing here, removing the 2 fallback `get_metadata()` functions. Any concerns/thoughts on how this would break compatibility?